src/service/cia301/co_csdo.c

```diff
@@ -206,6 +206,7 @@ static CO_ERR COCSdoUploadExpedited(CO_CSDO *csdo)
 
     cmd = CO_GET_BYTE(csdo->Frm, 0u);
     width = 4u - ((cmd >> 2u) & 0x03u);
+    csdo->Tfer.Size = width;
```

Comment:
Not sure this is correct, as csdo->Tfer.Size seems to be to size of the buffer, but is wrongly defined in the header of COCSdoRequestUpload:
```
 *
 * \param buf
 *   Reference to data to be downloaded to server
 *
 * \param size
 *   Size of downloaded data
 *
```

Reply:
Normally when I send a segmented transfer, the size variable gets set for me and I can use it to properly trim the initially large buffer I have. without this line, I was getting back the size of my entire upload buffer back, not the size of the transfer. I am happy to make a separate upload expedited function that I call in these circumstances instead of the normal COCSdoUploadExpedited function that achieves this. But either way, I do need this size setting because that is what I end up using on the callback side in Rust

Reply:
In that case I think `csdo->Tfer.Size = width;` should be moved to line 213

Reply:
copy that. Addressed and changed